server/util/flagutil/yaml/yaml.go

```diff
@@ -543,39 +559,40 @@ func onSIGHUP(fn func() error) {
 	}()
 }
 
-// PopulateFlagsFromData takes the path to some YAML file, reads it, and
-// unmarshals it, then uses the umnarshaled data to populate the unset flags
-// with names corresponding to the keys.
+// PopulateFlagsFromFile populates unset flags with data from an input config file
 func PopulateFlagsFromFile(configFile string) error {
 	log.Infof("Reading buildbuddy config from '%s'", configFile)
-
-	_, err := os.Stat(configFile)
-
-	// If the file does not exist then skip it.
-	if os.IsNotExist(err) {
-		log.Warningf("No config file found at %s.", configFile)
-		return nil
-	}
-
-	fileBytes, err := os.ReadFile(configFile)
+	fileBytes, err := file.ReadFile(configFile)
 	if err != nil {
-		return fmt.Errorf("Error reading config file: %s", err)
+		// If the file does not exist then skip it.
+		if status.IsNotFoundError(err) {
+			log.Warningf("No config file found at %s.", configFile)
+			return nil
+		}
+		return err
 	}
 
-	if err := PopulateFlagsFromData(fileBytes); err != nil {
+	return PopulateFlagsFromData(fileBytes)
+}
+
+// PopulateFlagsFromFileWithReread calls PopulateFlagsFromFile, and ensures the flags are reset
+// if the connection is disconnected
+func PopulateFlagsFromFileWithReread(configFile string) error {
+	err := PopulateFlagsFromFile(configFile)
+	if err != nil {
 		return err
 	}
 
-	// Setup a listener to re-read the config file on SIGHUP.
+	RereadFlagsOnDisconnect(configFile)
+	return nil
+}
+
+// RereadFlagsOnDisconnect sets up a listener to re-read a config file and reset flags on SIGHUP
+func RereadFlagsOnDisconnect(configFile string) {
 	onSIGHUP(func() error {
-		log.Infof("Re-reading buildbuddy config from '%s'", configFile)
-		fileBytes, err := os.ReadFile(configFile)
-		if err != nil {
-			return fmt.Errorf("Error reading config file: %s", err)
-		}
-		return PopulateFlagsFromData(fileBytes)
+		log.Infof("Re-reading buildbuddy config %s", configFile)
+		return PopulateFlagsFromFile(configFile)
```

Comment:
We can't just re-ingest a config file because:

1. The values from the previous read of the config file will still be set, so if you remove a definition in the YAML it will not be the flag default and will instead be the old value.

1. Slice flags work by appending new values encountered instead of replacing them, so even if you have the same config file, all slice definitions will append their values to the flag again.

If you want to do this, we need to make a `Clear` function for flags to allow us to clear them, then re-parse flags, then re-ingest the config file.